SerialPrograms/Source/PokemonSV/Inference/Overworld/PokemonSV_OliveDetector.cpp

```diff
@@ -0,0 +1,386 @@
+/*  Olive Detector
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+#include "Kernels/Waterfill/Kernels_Waterfill_Session.h"
+#include "CommonFramework/Globals.h"
+#include "CommonFramework/Exceptions/OperationFailedException.h"
+#include "CommonFramework/Exceptions/OliveActionFailedException.h"
+#include "CommonFramework/VideoPipeline/VideoOverlayScopes.h"
+#include "CommonFramework/ImageMatch/ImageDiff.h"
+#include "CommonFramework/ImageMatch/ExactImageMatcher.h"
+#include "CommonFramework/ImageTools/BinaryImage_FilterRgb32.h"
+#include "CommonFramework/ImageTools/WaterfillUtilities.h"
+#include "CommonFramework/ImageMatch/WaterfillTemplateMatcher.h"
+#include "CommonFramework/ImageTools/ImageFilter.h"
+#include "CommonFramework/Tools/ConsoleHandle.h"
+#include "NintendoSwitch/Programs/NintendoSwitch_SnapshotDumper.h"
+#include "PokemonSV/Inference/Overworld/PokemonSV_DirectionDetector.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "PokemonSV_OliveDetector.h"
+
+#include <iostream>
+using std::cout;
+using std::endl;
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+
+class OliveMatcher : public ImageMatch::WaterfillTemplateMatcher{
+public:
+    OliveMatcher() : WaterfillTemplateMatcher(
+        "PokemonSV/Olive.png", Color(0,0,0), Color(255, 255, 255), 5
```

Comment:
How is this supposed to work? The olive can in be multiple orientations as well as being eclipsed or occulted.

Reply:
All non-green pixels are filtered out. Then the largest blob of green is considered to be the Olive. Unlike all the other detectors, the RMSD doesn't matter much. This is why it doesn't care if the olive is eclipsed or in a different orientation, as long as it's the largest blob of green.